packages/core/src/render3/view.ts

```diff
@@ -0,0 +1,367 @@
+    return getEmbeddedViewFactoryInternal<T>(declarationTNode, declarationLView) as any;
+  }
+  return null;
+}
+
+/**
+ * The internal implementation for {@link getEmbeddedViewFactory}.
+ *
+ * @param declarationTNode The template node where the embedded template was declared
+ * @param declarationLView The local view where the embedded template was declared
+ */
+export function getEmbeddedViewFactoryInternal<T extends{}>(
+    declarationTNode: TNode, declarationLView: LView): EmbeddedViewFactoryInternal<T>|null {
+  const templateTView = declarationTNode.tViews;
+  if (templateTView) {
+    if (Array.isArray(templateTView)) {
+      // We are currently not supporting inline views.
+      throw new Error('Array of TViews not supported');
+    }
+
+    return function embeddedViewFactory(context: T) {
+      const _isParent = getIsParent();
+      const _previousOrParentTNode = getPreviousOrParentTNode();
+      const _useIvyAnimationCheck = shouldUseIvyAnimationCheck();
+      const _addingEmbeddedRootChild = _useIvyAnimationCheck && getAddingEmbeddedRootChild();
+      try {
+        setIsParent(true);
+        setPreviousOrParentTNode(null !);
+        setAddingEmbeddedRootChild(true);
+
+        const lView = createLView(
+            declarationLView, templateTView, context, LViewFlags.CheckAlways, null, null);
+
+        ngDevMode && assertEqual(
+                         declarationTNode.type, TNodeType.Container,
+                         'TNode type should be Container for embedded views');
+
+        const declarationContainer = unwrapLContainer(declarationLView[declarationTNode.index]) !;
+        ngDevMode && assertLContainer(declarationContainer);
+        const declarationQueries = declarationContainer[QUERIES];
+        if (declarationQueries) {
+          lView[QUERIES] = declarationQueries.createView();
+        }
+
+        assignTViewNodeToLView(templateTView, null, -1, lView);
+
+        if (templateTView.firstTemplatePass) {
+          templateTView.node !.injectorIndex = declarationTNode.injectorIndex;
+        }
+
+        renderEmbeddedTemplate(lView, templateTView, context);
+
+        return lView;
+      } finally {
+        _useIvyAnimationCheck && setAddingEmbeddedRootChild(_addingEmbeddedRootChild);
+        setIsParent(_isParent);
+        setPreviousOrParentTNode(_previousOrParentTNode);
+      }
+    };
+  }
+  return null;
+}
+
+/**
+ * Gets or creates an Angular view container from a DOM node
+ *
+ * Gets or creates a {@link ViewContainer} instance from a container comment DOM node
+ * (`<!-- container -->`), this is something generally added to the DOM by template instructions and
+ * other mechanisms that use embedded views such as {@link ngIf} or {@link ngForOf}.
+ * @param containerComment The DOM node to use to find the view container instance.
+ */
+export function getViewContainer(containerComment: RComment): ViewContainer|null {
```

Comment:
Is the type of this arg correct?  Here and in the docs comment above, it implies that only comments should be supported, but you have a test [here](https://github.com/angular/angular/pull/29339/files#diff-9c4a5bd39a10cbff63fa0ffbd9a656fdR60) that passes in a `<div>` element.

If we expect this to work with non-comment elements (and already have a test for it), we should update the function signature and docs comment.

```suggestion
export function getViewContainer(containerHost: RElement | RComment): ViewContainer|null {
```

Reply:
Yep, this was my question as well....

Reply:
Actually changed to `containerNative` to align with where it is stored in `LContainer`.